Migrate events docs to notifications

nikolajlauridsen opened the pull request:
This migrates the Events docs and sub-pages: https://our.umbraco.com/documentation/Reference/Events/

I've also updated the events mentioned in the trees docs to notifications instead.

While I was migrating this I also noted that the Events section in the MemberService page on the v8 docs (https://our.umbraco.com/documentation/Reference/Events/MemberService-Events/) is wrong, it looks like a copy-paste job gone bad, for instance, the "DeletingVersions" event doesn't exist, and the description column also mentions "Id: Gets the id of the IContent object being deleted.", which of course doesn't make sense, it's the same for a lot of the other events as well.

I also think it would make sense to add an "extending notifications" doc about creating and publishing your own custom notifications, but sadly I'm a little pressed for time to do it right now.

I initially made this for the wrong repo, oops 🙈, so I've had to change the filenames and links to this format, I think I got them all though, I have however split the ContentTypeService section into multiple, ContentTypeService, MediaTypeService and MemberTypeService, since these are three different services, that just inherits from the same base, so it made more sense to me for them to have their own pages.

sofietoft commented:
Thanks for moving this over here instead @nikolajlauridsen ! 💪 Much appreciated.

I'll make sure we review asap!
And about those warnings... it make sense that the items start with lowercase in these labels, so don't worry about it. We can still merge it 😁

sofietoft reviewed (commented):
Looks great @nikolajlauridsen !
I've made a couple of suggestions, and then I've also taken the liberty to fix some broken links 💪 

I was also thinking... Everything that was "Events" before in v8, is now in v9 going to be called "Notifications" - right?
In that case, does it still make sense to have all these new articles in the Reference/Events section? 🤔 
I mean, we could move all of them to a new Reference/Notification section instead, and then add notes to the old Events articles about where to find the v9 equivalent... What do you think of that idea?

And just for the record; I'm not saying I want you to do it 😅 I'm just thinking about ideas on how best we can solve this "name change" in the docs 🤔

nikolajlauridsen commented on Reference/Events/EditorModel-Events/Customizing-the-links-box-v9.md:
````diff
@@ -0,0 +1,26 @@
+
+With the `SendingContentNotification` event, we can manipulate the links in the `Urls` property - e.g. replace it with some custom links (although a URL provider would be more suitable):
+
+```C#
+public void Handle(SendingContentNotification notification)
+{
+    notification.Content.Urls = new[]
+    {
+        new UrlInfo($"/products/?id={notification.Content.Id}", true, CultureInfo.CurrentCulture.Name)
+    };
+}
+```
````
No the semicolon is in the right place, it's because it's a collection initializer that has a bit of a weird syntax 😄 

The curly brackets are more like a parenthesis, in this case, it becomes a bit more apparent when you add more than one item, then it would look like: 

```C#
notification.Content.Urls = new[]
{
    new UrlInfo($"/products/?id={notification.Content.Id}", true, CultureInfo.CurrentCulture.Name),
    new UrlInfo($"/SomeOtherLink/?id={notification.Content.Id}", true, CultureInfo.CurrentCulture.Name)
};
```
If it makes sense? 😄

sofietoft replied:
Aah! Well, I learned something new then 💪 
Thanks for explaining!

nikolajlauridsen commented:
> I was also thinking... Everything that was "Events" before in v8, is now in v9 going to be called "Notifications" - right?
> In that case, does it still make sense to have all these new articles in the Reference/Events section? 🤔
> I mean, we could move all of them to a new Reference/Notification section instead, and then add notes to the old Events articles about where to find the v9 equivalent... What do you think of that idea?

Yep, that's correct, the CMS no longer raises any events, we only use notifications instead.

I think it's a good idea to move them to a new notification section instead like you said since it does feel a bit weird that they're under the events section. When I made the PR to UmbracoCmsDocs I had renamed the files to notifications instead because it makes more sense, the reason why I renamed them back to Events was just to make it apparent that they replace the events documentation 😃

sofietoft commented:
Great! 💪 I'll make sure to add a task on moving these new articles!
